internal/blocksync/reactor.go

```diff
@@ -465,7 +465,10 @@ func (r *Reactor) poolRoutine(ctx context.Context, stateSynced bool, blockSyncCh
 			switch {
 			// TODO(sergio) Might be needed for implementing the upgrading solution. Remove after that
 			//case state.LastBlockHeight > 0 && r.store.LoadBlockExtCommit(state.LastBlockHeight) == nil:
-			case state.LastBlockHeight > 0 && blocksSynced == 0:
+			case state.LastBlockHeight > 0 &&
+				state.ConsensusParams.ABCI.VoteExtensionsEnabled(state.LastBlockHeight) &&
+				// can you switch to consensus if no blocks are sync'd but you already have a ext commit?
```

Comment:
Is this a question to the reviewers?
My answer to that is, yes. But AFAIU, if the whole condition in (old) line 468 is true, that won't be the case (I was assuming that a crashed-recovered node doesn't do block-sync...)
The commented out condition in line 467 was my dumber, less sexy attempt... but which probably matches what you mean by this comment.

Reply:
Had originally put this comment, elsewhere, but actually it makes more sense here:

I see what this was doing, got it. So, it is possible to blocksync a crash recovered node. I've added an updated check along with documentation that I think will make this a bit cleaner without having to check the db on each iteration of this loop, let me know if it makes sense to you.

Reply:
Thanks. I went through the new version and made a suggestion to simplify it.